home/widgets.py

```diff
@@ -163,3 +163,53 @@ def _observe_enabled(self, change):
             )
         else:
             self.value = ""
+
+
+class LogOutputWidget(ipw.VBox):
+
+    value = traitlets.Unicode()
+
+    def __init__(self, num_min_lines=10, **kwargs):
+        self._num_min_lines = num_min_lines
+        self._output = ipw.HTML(layout=ipw.Layout(min_width="60em"))
+        self._refresh_output()
+        super().__init__(
+            children=[
+                self._output,
+            ],
+            **kwargs,
+        )
+
+    def write(self, text):
+        self.value += text
+
+    @traitlets.default("value")
+    def _default_value(self):
+        if self._num_min_lines > 0:
+            return "\n" * self._num_min_lines
+        return ""
+
+    @traitlets.observe("value")
+    def _refresh_output(self, _=None):
+        with self.hold_trait_notifications():
+            self._output.value = self._format_output(self.value)
+
+    def _format_output(self, text):
+        lines = text.splitlines()
+
+        # Add empty lines to reach the minimum number of lines.
+        lines += [""] * max(0, self._num_min_lines - len(lines))
+
+        # Replace empty lines with single white space to ensure that they are actually shown.
+        lines = [line if len(line) > 0 else " " for line in lines]
+
+        # Replace the first line if there is no output whatsoever
+        if len(text.strip()) == 0 and len(lines) > 0:
+            lines[0] = "[waiting for output]"
+
+        text = "\n".join(lines)
+        return (
+            f"""Intalling dependencies... </br> <pre style="background-color: #1f1f2e; color: white; line-height: 100%">{text}</pre>"""
```

Comment:
First of all, there is a typo:
```suggestion
            f"""Installing dependencies... </br> <pre style="background-color: #1f1f2e; color: white; line-height: 100%">{text}</pre>"""
```
But more importantly, this specific message should not go into a generic widget. I suggest to add a "template" trait with some kind of reasonable default and then use it like this:
```suggestion
            self.template.format(text=text)
```
For the specific application you can then replace the template to add the "Installing dependencies..." line.